### PR TITLE
Kd/use factor graphs

### PR DESCRIPTION
**Pull request by kdomino**

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -14,6 +14,7 @@ LowRankApprox = "898213cb-b102-5a47-900c-97e73b919f73"
 MetaGraphs = "626554b9-1ddb-594c-aa3c-2596fe9399a5"
 Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
 Requires = "ae029012-a4dd-5104-9daa-d747884805df"
+Revise = "295af30f-e4ad-537b-8983-00126c2a3abe"
```
Comment (lpawela) [inline Project.toml]: Do not add `Revise` to deps

File: src/utils.jl [hunk 2]
```diff
@@ -2,10 +2,17 @@ export idx, ising, proj
 export HadamardMPS, rq
 export all_states, local_basis, enum, state_to_ind, rank_vec
 export @state
+export reshape_row
 
 using Base.Cartesian
 import Base.Prehashed
 
+function reshape_row(A::AbstractArray{T}, dims::Tuple) where {T <: Number}
```
Comment (lpawela) [inline src/utils.jl]: Why is this here? Is it used in the code somewhere?

File: test/test_helpers.jl [hunk 3]
```diff
@@ -52,43 +52,80 @@ function make_interactions_case1()
     L = 9
 
     D = Dict{Tuple{Int64,Int64},Float64}()
-    push!(D, (1, 1) => 1.0)
+    push!(D, (1,1) => 1.0)
```
Comment (lpawela) [inline test/test_helpers.jl]: Use a single `push!` like
```julia

push!(
D, 
(2,2) => -2.0,
(3,3) => 4.0,
(4,4) => 0.0
)

```

File: test/utils.jl [hunk 4]
```diff
@@ -3,4 +3,25 @@
     d = 2^length(σ)
     x = rand(2, 2, d)
     @test (@state x[1, 1, σ]) == x[1, 1, 20]
+end
+
+@testset "Reshape by rows" begin
```
Comment (lpawela) [inline test/utils.jl]: Again, why do we need this reshape?

**Review by lpawela — CHANGES_REQUESTED** [review]

File: Project.toml (hunk 1 above)
Comment (kdomino) [reply Project.toml]: removed

File: src/utils.jl (hunk 2 above)
Comment (kdomino) [reply src/utils.jl]: removed

File: test/test_helpers.jl (hunk 3 above)
Comment (kdomino) [reply test/test_helpers.jl]: corrected

File: test/utils.jl (hunk 4 above)
Comment (kdomino) [reply test/utils.jl]: removed in tests

**Review by lpawela — APPROVED** [review]
